projects/step-core/src/lib/directives/recursive-tab-index.directive.ts

```diff
@@ -0,0 +1,63 @@
+import { AfterViewInit, Directive, ElementRef, Input } from '@angular/core';
+
+declare global {
+  interface Array<T> {
+    bfsFilter: <T>({ predicate, children }: { predicate?: (item: T) => boolean; children: (item: T) => T[] }) => T[];
+  }
+}
+
+/**
+ * The time complexity of a Breadth-First Search (BFS) algorithm is O(|V| + |E|),
+ * where |V| is the number of vertices and |E| is the number of edges in the graph being traversed.
+ * The reason for this is that in the worst-case scenario, BFS needs to visit all vertices and all edges.
+ */
+Array.prototype.bfsFilter = function bfsFilter<T>({
```

Comment:
In my opinion this function shouldn't be included in Array's prototype.
It is used to walk through the tree structures. 
But array isn't a tree.